pulp_file/tests/functional/api/test_generic_list.py

```diff
@@ -35,3 +35,34 @@ def test_read_all_content_guards_generic(
     assert response.count != 0
     for content_guard in response.results:
         assert content_guard.name is not None
+
+
+@pytest.mark.parallel
+def test_read_all_master_model_remotes_generic(
+    remotes_api_client, gen_object_with_cleanup, file_remote_api_client
+):
+    remote_data = {"name": str(uuid.uuid4()), "url": "http://example.com"}
+    remote1 = gen_object_with_cleanup(file_remote_api_client, remote_data)
+    remote_data = {"name": str(uuid.uuid4()), "url": "http://example.org"}
+    remote2 = gen_object_with_cleanup(file_remote_api_client, remote_data)
+
+    response = remotes_api_client.list()
+    assert response.count != 0
+    r1_found = False
+    r2_found = False
+
+    for remote in response.results:
+        if remote.pulp_href == remote1.pulp_href:
+            r1_found = True
+        elif remote.pulp_href == remote2.pulp_href:
+            r2_found = True
+
+    assert r1_found and r2_found
+
+    """url = "https://" + gethostname() + "/pulp/api/v3/remotes/"
+    response = requests.get(url=url)
+    parsed_response = loads(response.text)
+    hrefs = []
+    for result in parsed_response["results"]:
+        hrefs.append(result["pulp_href"])
+    assert remote1.pulp_href in hrefs and remote2.pulp_href in hrefs"""
```

Comment:
Maybe you can rephrase the asserts above to this line again. It's way more readable. Actually i would make it two asserts:
```suggestion
    assert remote1.pulp_href in hrefs
    assert remote2.pulp_href in hrefs
```